### DISCUSSION TITLE
Can someone clarify what the plan is with the deprecated MQTT topics?

### DISCUSSION THREAD

**Discussion by pjdubya** [teslamate-org/teslamate#3874] [Q&A] [answered]
Hi everyone. I'm new to teslamate but am excited to use it for some of my home automation routines. For one use case I'm working on, there are some MQTT topics that are currently marked 'DEPRECATED' but I couldn't see an explanation of why they are being removed, when they may be removed, and what alternatives there may be. It's these topics:

teslamate/cars/$car_id/active_route_destination
teslamate/cars/$car_id/active_route_latitude
teslamate/cars/$car_id/active_route_longitude

My use case is that when I'm away from home some other unrelated routines are active (eg thermostat in eco mode) that I will want to disable when I'm both away from home and ,my active_route_destination indicates I'm on my way to "Home". So I'm thinking of using this topic for that, but I would like to understand more why this is deprecated. I didn't see any discussion or notes in the code about it so can anyone point me to a clearer understanding?

**Comment by brianmay** [comment, marked as answer]
See `teslamate/cars/$car_id/active_route` - it outputs everything as a json blob.

**Reply by pjdubya** [reply]
Interesting, thanks for the quick help. I'm seeing the 3 deprecated topics getting published to, but nothing yet on `teslamate/cars/$car_id/active_route` . I checked and I'm running the :latest aka 1.28.5.  I don't suppose I need to do something to enable the new topic so I can build my automation on that? I'll also just do some additional test drives with navigation to see if it just hasn't shown up yet.

**Comment by brianmay** [comment]
The new topic may not have been included yet in the latest release of teslamate.

**Reply by pjdubya** [reply]
Yep, I see now that you're right. I pulled the latest code and ran it locally -- with that, the new topic gets published to as expected. I see that the new topic code didn't merge until a couple days after the last release, so that makes sense. Will just build my logic then to expect this topic in the next release. Thanks for the info and all the hard work to make this data available in easy to use formats. The new JSON object looks nicely extensible and minutes_to_arrival to already exactly the field I need, so that's fantastic.
